storybookjs/storybook
# CLI: Add `--no-dev` option to `init`

Closes #25510





## What I did



adds optional `--dev` and `--no-dev` options to the `storybook init` CLI.
- this option allows users of the `init` CLI to force the init workflow to conclude before running the dev server after initialization.
- this is still the default behavior when `storybook init` is run in CI or in the Storybook sandbox, although the option can also be used to override the behavior in those cases as well.
   - unfortunately, because Commander automatically initializes the boolean with the default value of `true`, and I couldn't override the default value from the negated `--no-dev` option, I was forced to add a `--dev` option with the correct defaults. see https://github.com/tj/commander.js/issues/2068#issuecomment-1804524585 and https://github.com/tj/commander.js/issues/979 for more on this topic.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

_as far as I could tell, there aren't automated tests for the `init` CLI?_

#### Manual testing

Ran the sandbox as described. Additionally, I ran the following against a fresh `create-next-app` to test that the flag works as expected, that the default behaviors still worked as expected, and that the intersection of the defaults and the flags also worked.

```sh
$ > npx create-next-app

$ > node ../storybook/code/lib/cli/bin/index.js init
$ > node ../storybook/code/lib/cli/bin/index.js init --dev
$ > node ../storybook/code/lib/cli/bin/index.js init --no-dev

$ > CI=true node ../storybook/code/lib/cli/bin/index.js init
$ > CI=true node ../storybook/code/lib/cli/bin/index.js init --dev
$ > CI=true node ../storybook/code/lib/cli/bin/index.js init --no-dev

$ > IN_STORYBOOK_SANDBOX=true node ../storybook/code/lib/cli/bin/index.js init
$ > IN_STORYBOOK_SANDBOX=true node ../storybook/code/lib/cli/bin/index.js init --dev
$ > IN_STORYBOOK_SANDBOX=true node ../storybook/code/lib/cli/bin/index.js init --no-dev
```

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
